hlapipeline/alignimages.py

```diff
@@ -552,27 +599,63 @@ def update_image_wcs_info(tweakwcs_output,imagelist):
 
     Returns
     -------
-    Nothing!
+    out_file_list : list
+        a list of the updated images files if wcs_dest = 'header', or a list of the newly created headerlet files if wcs_dest = 'headerlet'.
     """
+    out_image_list=[]
+    out_headerlet_list=[]
     imgctr = 0
     for item in tweakwcs_output:
         if item.meta['chip'] == 1:  # to get the image name straight regardless of the number of chips
             image_name = imagelist[imgctr]
             if imgctr > 0: #close previously opened image
-                print("CLOSE {}".format(hdulist[0].header['FILENAME'])) #TODO: Remove before deployment
+                imageName = hdulist[0].header['FILENAME']
+                print("CLOSE {}\n".format(imageName)) #TODO: Remove before deployment
                 hdulist.flush()
                 hdulist.close()
+                out_image_list.append(imageName)
+                # Create headerlet
+                out_headerlet = headerlet.create_headerlet(imageName, hdrname=wcsName, wcsname=wcsName)
+
+                # TODO: First of two calls to Michele's update headerlet subroutine goes here!
+
+                if imageName.endswith("flc.fits"):
+                    headerlet_filename = imageName.replace("flc", "flt_hlet")
+                if imageName.endswith("flt.fits"):
+                    headerlet_filename = imageName.replace("flt", "flt_hlet")
+                out_headerlet.writeto(headerlet_filename,clobber=True)
+                print("Wrote headerlet file {}.\n\n".format(headerlet_filename))
+                out_headerlet_list.append(headerlet_filename)
             hdulist = fits.open(image_name, mode='update')
+            #generate wcs name for updated image header, headerlet
+            if not hdulist['SCI',1].header['WCSNAME'] or hdulist['SCI',1].header['WCSNAME'] =="": #Just in case header value 'wcsname' is empty.
+                wcsName = "FIT_{}".format(item.meta['catalog_name'])
+            else:
+                wcsName = "{}-FIT_{}".format(hdulist['SCI',1].header['WCSNAME'],item.meta['tweakwcs_info']['catalog'])
```

Comment:
This would result in a WCSNAME like 'IDC_0461802ej-GSC240-FIT_GAIADR2' for data which has already been updated with the a priori solutions.  Instead, we need to simply replace '-GSC240' (in this example) with 'FIT_GAIADR2'.  This can be done using: 
wname = hdulist['sci',1].header['wcsname']
wcsName = "{}-FIT_{}'.format(wname[:wname.index('-')], item.meta['tweakwcs_info']['catalog'])

Reply:
Resolved in commit [0211dc1](https://github.com/spacetelescope/hlapipeline/pull/56/commits/0211dc10040c1e4ac548e693076b4ee4609d8194)